tests/tests_integration/test_api/test_data_workflows.py

```diff
@@ -640,5 +589,24 @@ def test_trigger_run_history(
         history = cognite_client.workflows.triggers.get_trigger_run_history(
             external_id=workflow_scheduled_trigger.external_id
         )
-        # it would take too long to wait for the trigger to run, so we just check that the history is not None
-        assert history is not None
+        assert len(history) > 0
+        assert history[0].external_id == workflow_scheduled_trigger.external_id
+        assert history[0].workflow_external_id == workflow_scheduled_trigger.workflow_external_id
+        assert history[0].workflow_version == workflow_scheduled_trigger.workflow_version
+
+        detailed = cognite_client.workflows.executions.retrieve_detailed(history[0].workflow_execution_id)
+        assert detailed is not None
+        assert detailed.metadata == workflow_scheduled_trigger.metadata
+        # version gets appended to input when executed by a trigger
+        workflow_scheduled_trigger.input["version"] = workflow_scheduled_trigger.workflow_version
+        assert detailed.input == workflow_scheduled_trigger.input
+
+    @pytest.mark.xfail(reason="Non-existing trigger run history returns [] instead of None")
```

Comment:
Has the WF team said they'll fix this? If not, lets remove the `xfail` and modify the test

Reply:
No, but I'll point it out to them and make the test pass the current API spec